### ISSUE TITLE
Add support for memory optimised tables

### ISSUE THREAD

**Issue by sjh37** [sjh37/efreversepoco#782]
The first step is ensuring you are on compatibility level >=130. Run this query to find out the current compatibility level:

```sql
SELECT compatibility_level FROM sys.databases WHERE name = DB_NAME();
```

Check if In-Memory OLTP is supported for this server edition and database pricing tier. This has to be 1.
```sql
SELECT CAST(SERVERPROPERTY(N'IsXTPSupported') AS BIT) AS IsXTPSupported
```

Next, list tables where this is switched on:

```sql
SELECT SCHEMA_NAME(schema_id) SchemaName, name TableName
FROM sys.tables
WHERE is_memory_optimized = 1;
```

For any memory-optimised tables

```c#
-- EF Core 6
modelBuilder
    .Entity<Blog>()
    .IsMemoryOptimized();

-- EF Core 7
modelBuilder
    .Entity<Blog>()
    .ToTable(b => b.IsMemoryOptimized());
```

More info [microsoft](https://learn.microsoft.com/en-us/sql/relational-databases/in-memory-oltp/overview-and-usage-scenarios?view=sql-server-ver16#sample-script) and [red-gate](https://www.red-gate.com/simple-talk/databases/sql-server/t-sql-programming-sql-server/beginner-guide-to-in-memory-optimized-tables-in-sql-server) and [github](https://github.com/microsoft/sql-server-samples/blob/master/samples/features/in-memory-database/in-memory-oltp/t-sql-scripts/enable-in-memory-oltp.sql)

**Comment by sjh37** [comment]
Completed

**Comment by sjh37** [comment]
Released in v3.8.1
